### DISCUSSION TITLE
input type of user_model in  BeanieUserDatabase mismatch

### DISCUSSION THREAD

**Discussion by asaah18** [fastapi-users/fastapi-users#1212] [Q&A] [answered]
<img width="811" alt="image" src="https://github.com/fastapi-users/fastapi-users/assets/32853027/89f28f8b-546e-40cc-9114-2a175645e90f">
PyCharm is complaining about type mismatch between expected and given type of `user_model` argument in `BeanieUserDatabase `. 

Just to be clear, the **code works perfectly**, it's just that PyCharm is complaining.

FastAPI-users version: 11.0.0

**Comment by frankie567** [comment, marked as answer]
That's indeed a typing error. We make heavy use of generics, so if you write this:

```py
async def get_user_db():
    yield BeanieUserDatabase[User](User)
```

PyCharm should be happy.
